spark-3.1-spanner-lib/src/test/java/com/google/cloud/spark/spanner/SparkFilterUtilsTest.java

```diff
@@ -76,50 +54,42 @@ public void testValidFilters() {
                 StringStartsWith.apply("foo", "abc"),
                 StringEndsWith.apply("foo", "def"),
                 StringContains.apply("foo", "abcdef")));
-    if (dataFormat != ARROW || pushAllFilters) {
-      // If pushAllFilters isn't true, the following filter can't be handled for ARROW.
-      validFilters.add(Or.apply(IsNull.apply("foo"), IsNotNull.apply("foo")));
-    }
     validFilters.forEach(
-        f ->
-            assertThat(SparkFilterUtils.unhandledFilters(pushAllFilters, dataFormat, f)).isEmpty());
+        f -> assertThat(SparkFilterUtils.unhandledFilters(pushAllFilters, f)).isEmpty());
   }
 
   @Test
   public void testMultipleValidFiltersAreHandled() {
     Filter valid1 = EqualTo.apply("foo", "bar");
     Filter valid2 = EqualTo.apply("bar", 1);
-    assertThat(SparkFilterUtils.unhandledFilters(pushAllFilters, dataFormat, valid1, valid2))
-        .isEmpty();
+    assertThat(SparkFilterUtils.unhandledFilters(pushAllFilters, valid1, valid2)).isEmpty();
   }
 
   @Test
   public void testInvalidFilters() {
+    if (true) {
+      // TODO: Some of these code paths are unused but for the sake of
```

Comment:
It's not working because we should keep pushAllFilters logic. This can be done in the followup changes though.